### PR TITLE
Docs corrections

### LINKED ISSUE #51: Save / refresh flow
Hi,
I have problem with refreshing / saving flow.

Someone has saved file on the server, I would like to save this file from vim.

When I'm trying to save `ApexSave` file I get following error:
    
    Operation failed
    Outdated file(s) detected.`
    xxx.cls => Modified By: xxx; at: 2015-09-21T11:17:53.000Z; Local version saved at: 2015-09-18T10:50:25.000Z`
    Use 'refresh' before 'deploy'.
Then when i go with `ApexRefreshFile`
    
    Remote version is saved in: /home/xxx/force.com-projects/apex/vim-deployment/diffWithRemote/src
    No differences between local version and remote Org detected

After this, when i try to save with `ApexSave` I'm getting the same error as the first one.

What should to be able to save this file on the server?

### PR DESCRIPTION
**Pull request by kgrzywacz**
Reviewed apex-commands section in documentation, added descriptions for few commands, ordered command alphabetically, corrected some errors. Corrected two misspelled words in ApexStage's comments.

This closes #51 .

### CODE REVIEW

File: doc/force.com.txt [hunk 1]
```diff
@@ -461,10 +446,226 @@ Params:
...
+Invoke metadata types list "menu" and load data of specified types.  
+Usually :|ApexRefreshProject| will retrieve all types defined in package.xml.
+:|ApexRetrieve| can be used to load types which are not part of current
+package.xml Use this command when package.xml does not yet contain necessary
+metadata type(s). If retrieved type is not part of the package.xml system will
+offer to update package.xml accordingly. 
+*WARNING* :|ApexRetrieve| command is Experimental. You may want to backup your
+project before running :|ApexRetrieve|.
+
+There are 2 potential scenarios which may involve :|ApexReteieve|
```
Comment (chexxor) [inline doc/force.com.txt]: Typo: "ApexReteieve" => "ApexRetrieve"

File: doc/force.com.txt [hunk 2]
```diff
@@ -461,10 +446,226 @@ Params:
...
+	:ApexRetrieve
+< then select CustomObjects line and then run >
+	:Expand
+< then select Account and run >
+	:Retrieve
+<
+                                                                      *ApexSave*
+
+Similar to :|ApexDeploy| but uses Tooling API instead of Metadata API.
+Takes modified files and tries to save them to SFDC.
```
Comment (chexxor) [inline doc/force.com.txt]: Would be nice to have "modified" defined here. File timestamp compared to SFDC timestamp? Does plugin persist a last-uploaded time for that file somewhere? Is a hash of that file calculated and compared to the hash of the last time it was uploaded?

**Comment by chexxor** [comment]
Thanks for caring for the docs! Alphabetical organization is a good idea, imo.

**Comment by neowit** [comment]
Thank you @kgrzywacz !

@chexxor - Local file checksum (CRC32), local timestamp and remote LastModifiedDate are saved in `.vim-force.com/session.properties`. Before every Save/Deploy command (without` <bang>`) we check if locally saved LastModifiedBy still the same as remote LastModifiedBy. If they are different then conflict is reported.

**Pull request merged**